chore: update readme for automated inclusion (#39)
open-feature/rust-sdk: README.md

<!-- x-hide-in-docs-end -->
<!-- The 'github-badges' class is used in the docs -->
<p align="center" class="github-badges">

>   <a href="https://github.com/open-feature/spec/releases/tag/v0.7.0">

Author: ```suggestion
  <a href="https://github.com/open-feature/spec/releases/tag/v0.5.2">
```

>   <a href="https://github.com/open-feature/spec/releases/tag/v0.5.2">

Author: This SDK isn't quite yet `0.6.0` compliant yet, though it's somewhat close. To me it seems `0.5.2` compliant.

Reviewer: I think the main missing features are hooks and events right? Am I missing anything?

Reviewer: Oh, and logging...

Author: I think that's the bulk of it.

FYI we had a lot of interest at KubeCon, and the Rust SDK came up a few times, so we may have some extra help incoming :crossed_fingers:

open-feature = "0.1.2"
```
<!-- x-release-please-end -->
### Usage

```rust

> async fn example() {
>     // Acquire an OpenFeature API instance.
>     let mut api = OpenFeature::singleton_mut().await;
> 
>     // configure a provider
>     api.set_provider(NoOpProvider::builder().bool_value(true).build())
>         .await;
> 
>     // create a client
>     let client = api.create_client();
> 
>     // get a bool flag value
>     let is_feature_enabled = client
>         .get_bool_value("v2_enabled", None, None)
>         .await
>         .unwrap_or(false);
> }

Reviewer 2: ```suggestion
async fn example() -> Result<(), Error> {
    // Acquire an OpenFeature API instance.
    let mut api = OpenFeature::singleton_mut().await;

    // configure a provider
    api.set_provider(NoOpProvider::new())
        .await;

    // create a client
    let client = api.get_client();

    // get a bool flag value
    let is_feature_enabled = client
        .get_bool_value("v2_enabled", false, None)
        .await;

    Ok(())
}
```

> ```

Author: We use this exact same example (flag key / type) in all SDKs. Please make sure I haven't made any silly mistakes, my Rust knowledge is approaching zero.

Reviewer 2: ```suggestion
```rust
async fn example() -> Result<(), Error> {
    // Acquire an OpenFeature API instance.
    let mut api = OpenFeature::singleton_mut().await;

    // configure a provider
    api.set_provider(NoOpProvider::new())
        .await;

    // create a client
    let client = api.get_client();

    // get a bool flag value
    let is_feature_enabled = client
        .get_bool_value("v2_enabled", false, None)
        .await;

    Ok(())
}
```
```
